crates/biome_glob/src/lib.rs

```diff
@@ -1,31 +1,86 @@
-use biome_rowan::{TextRange, TextSize};
+//! biome_glob provides a glob and glob list with exceptions matching.
+//!
+//! ## Matching a path against a glob
+//!
+//! You can create a glob from a string using [core::str::FromStr::from_str] or the corresponding method `parse`.
+//! A glob can match against anything that can be turned into a [std::path::Path].
+//! This is for example the aqcse of strings.
+//!
+//! ```
+//! use biome_glob::Glob;
+//!
+//! let glob = "*.rs".parse::<Glob>().expect("correct glob");
+//! assert!(glob.is_match("lib.rs"));
+//! assert!(!glob.is_match("src/lib.rs"));
+//! ```
+//!
+//! ## Matching against multiple globs
+//!
+//! When a path is expected to be matched against several globs,
+//! you should compile the path into a [CandidatePath].
+//! [CandidatePath] may speed up matching against several globs.
+//!
+//! ```
+//! use biome_glob::{CandidatePath, Glob};
+//!
+//! let globs: &[Glob] = &[
+//!     "**/*.rs".parse().expect("correct glob"),
+//!     "**/*.txt".parse().expect("correct glob"),
+//! ];
+//!
+//! let path = CandidatePath::new(&"a/path/to/file.txt");
+//!
+//! assert!(globs.iter().any(|glob| path.matches(glob)));
+//! ```
+//!
+//! ## Matching against multiple globs and exceptions
+//!
+//! biome_glob supports negated globs, which are particularly useful for encoding exceptions.
+//! In the following example we accept all files in the `src` dierctory, except the ones ending with the `txt` extension.
+//!
+//! ```
+//! use biome_glob::{CandidatePath, Glob};
+//!
+//! let globs: &[Glob] = &[
+//!     "**/*.rs".parse().expect("correct glob"),
+//!     "!**/*.txt".parse().expect("correct glob"),
+//! ];
+//!
+//! let path = CandidatePath::new(&"a/path/to/file.txt");
+//!
+//! assert!(!path.matches_with_exceptions(globs));
+//! ```
+//!
+//! ## Supported syntax
+//!
+//! A Biome glob pattern supports the following syntaxes:
+//!
+//! - star `*` that matches zero or more characters inside a path segment
+//! - globstar `**` that matches zero or more path segments
+//! - Use `\*` to escape `*`
+//! - `?`, `[`, `]`, `{`, and `}` must be escaped using `\`.
+//!   These characters are reserved for future use.
+//! - Use `!` as first character to negate the glob
+//!
+//! A path segment is delimited by path separator `/` or the start/end of the path.
+//!
```

Comment:
I believe we should have a brief explanation of the features available by this crate